site/docs/relations/logical_relations.md

```diff
@@ -229,8 +229,11 @@ The join operation will combine two separate inputs into a single output, based
 | Left  | Return all records from the left input. For each cross input match, return a record including the data from both sides. For any remaining non-matching records from the left input, return the left record along with nulls for the right input. |
 | Right | Return all records from the right input. For each cross input match, return a record including the data from both sides. For any remaining non-matching records from the right input, return the right record along with nulls for the left input. |
 | Semi | Returns records from the left input. These are returned only if the records have a join partner on the right side. |
+| Right Semi | Returns records from the right input. These are returned only if the records have a join partner on the left side. |
 | Anti  | Return records from the left input. These are returned only if the records do not have a join partner on the right side. |
-| Single | Returns one join partner per entry on the left input. If more than one join partner exists, there are two valid semantics. 1) Only the first match is returned. 2) The system throws an error. If there is no match between the left and right inputs, NULL is returned. |
+| Right Anti  | Return records from the right input. These are returned only if the records do not have a join partner on the left side. |
+| Single | Return all records from the left input.  For each cross input match, return a record including the data from both sides.  For any remaining non-matching records from the left input, return the left record along with nulls for the right input.  If a record from the left input has more than one match in the right input then an error must be raised.  Useful for nested sub-queries where we need exactly one record in output (or throw exception).  See Section 3.2 of https://15721.courses.cs.cmu.edu/spring2018/papers/16-optimizer2/hyperjoins-btw2017.pdf for more information. |
```

Comment:
```suggestion
| Single | Return all records from the primary (left) input with no join expansion. If at least one record from the secondary (right) input matches the primary, return one arbitrary matching record from the secondary (right) input. For any primary (left) records without matching secondary (right) records, return the primary (left) record along with nulls for the secondary (right) input. Similar to a left outer join but only returns one right match at most. Useful for nested sub-queries where we need exactly one record in output (or throw exception).  See Section 3.2 of https://15721.courses.cs.cmu.edu/spring2018/papers/16-optimizer2/hyperjoins-btw2017.pdf for more information. |
```

Reply:
I'm ok with this.  Not a huge fan of introducing new terminology (primary, secondary) and its confusing to me how this actually solves the nested subquery problem (since it doesn't raise an error) but I agree that we should document how engines are actually working when they are in disagreement with the paper.

Reply:
I introduced it since the other one referred to primary but it wasn't defined above.

The more I think about this I think we should call this Left Single and Right Single. That would be more consistent with other names. Having Single and Right Single is just weird.

Reply:
I agree that it should be left single and right single.  But for the purposes of this PR I just want to normalize the types available in the two join types (essentially a non-breaking change).  A rename could break code so I would rather accomplish that in a later step.

Reply:
> A rename could break code so I would rather accomplish that in a later step.

I'm inclined to fix it now while we're already fixing the other stuff instead of punting it. If we're changing the definition of what SINGLE means (since it was vague before), that's basically a breaking spec change even if the protobuf change itself isn't breaking. 

Who wants to spend the time to come back and fix that? If we need to fix something, I'd prefer to just fix it while we're in that exact code already.